sherpa/csrc/endpoint.cc

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright      2022  (authors: Pingfeng Luo)
+ *
+ * See LICENSE for clarification regarding multiple authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "sherpa/csrc/endpoint.h"
+#include "sherpa/csrc/log.h"
+
+namespace sherpa {
+
+static bool RuleActivated(const EndpointRule& rule,
+                          const std::string& rule_name,
+                          int trailing_silence, int utterance_length) {
+  bool contain_nonsilence = utterance_length > trailing_silence;
+  bool ans = (contain_nonsilence || !rule.must_contain_nonsilence) &&
+      trailing_silence >= rule.min_trailing_silence &&
+      utterance_length >= rule.min_utterance_length;
+  if (ans) {
+      SHERPA_LOG(INFO) << "Endpointing rule " << rule_name
```

Comment:
I suggest using `DEBUG` or `TRACE`.
The default log level is `INFO` and it will print a lot of logs here.